src/components/Miscellaneous/Loading.js

```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import { ActivityIndicator } from "react-native";
+import { Container } from "../../styles/GeneralStyles";
+
+const Loading = () => {
+  return (
+    <Container>
+      <ActivityIndicator size="large" />
+    </Container>
```

Comment:
I think you misunderstood. What I meant is something like
```
const Loading = ({ isLoading, children }) => {
  return (
    <Container>
      {isLoading ? <ActivityIndicator size="large" /> : children}
    </Container>
  );
};
```
And to use it you simply wrap around the desired screen passing the "isLoading" state